mlora/executor/task/task.py

```diff
@@ -94,7 +116,40 @@ def _pre_dataset(self):
         if preprocess_type not in preprocess_func:
             raise NotImplementedError
 
-        data = preprocess_func[preprocess_type](data)
+        # If data preprocess_type is shuffle, create a cache folder,
+        # to store shuffled data and use it for saving checkpoints.
+        if preprocess_type == "shuffle":
```

Comment:
can we put the function into line 97 or create a new function?

Reply:
maybe it's better to avoid using the "if" branch here because different process_type have different process functions defined in line 94.

Reply:
OK, let me have a try.